libmapper/providers/azure/components/securitygroup.go

```diff
@@ -79,16 +79,64 @@ func (i *SecurityGroup) GetTag(tag string) string {
 
 // Diff : diff's the component against another component of the same type
 func (i *SecurityGroup) Diff(c graph.Component) bool {
+	cs, ok := c.(*SecurityGroup)
+	if ok {
+		if len(i.SecurityRules) != len(cs.SecurityRules) {
+			return true
+		}
+		for j := range cs.SecurityRules {
+			if i.SecurityRules[j].Name != cs.SecurityRules[j].Name {
+				return true
+			}
+			if i.SecurityRules[j].Description != cs.SecurityRules[j].Description {
+				return true
+			}
+			if i.SecurityRules[j].Protocol != cs.SecurityRules[j].Protocol {
+				return true
+			}
+			if i.SecurityRules[j].SourcePort != cs.SecurityRules[j].SourcePort {
+				return true
+			}
+			if i.SecurityRules[j].DestinationPortRange != cs.SecurityRules[j].DestinationPortRange {
+				return true
+			}
+			if i.SecurityRules[j].SourceAddressPrefix != cs.SecurityRules[j].SourceAddressPrefix {
+				return true
+			}
+			if i.SecurityRules[j].DestinationAddressPrefix != cs.SecurityRules[j].DestinationAddressPrefix {
+				return true
+			}
+		}
+	}
 
 	return false
 }
 
 // Update : updates the provider returned values of a component
 func (i *SecurityGroup) Update(c graph.Component) {
+	cs, ok := c.(*SecurityGroup)
+	if ok {
+		i.ID = cs.ID
+		var rules []securitygroup.SecurityRule
+		for j := range cs.SecurityRules {
+			rules = append(rules, securitygroup.SecurityRule{
```

Comment:
Not sure if this is needed, this stuff should only be mapped if its not user provided and returned by azure